### ISSUE TITLE
Add restartability/continuation to osrm-extract

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#3651]
The `osrm-extract` task takes a while, particularly on large datasets.  When used in unreliable computing environments (i.e. market-priced cloud providers), it's not uncommon for long-running tasks to get interrupted.

`osrm-extract` can be modified to safe its progress.

https://github.com/Project-OSRM/osrm-backend/blob/c48fc58eb28bfd296b9ca9483ba938b758ec0b9b/src/extractor/extractor.cpp#L245

This line represents approximately the halfway processing point for `osrm-extract`.

Through an serendipitous accident of apathy/not-enough-time, `osrm-extract` could re-start processing at L245 if it happens to get killed after that for some reason.

/cc @miccolis

**Comment by daniel-j-h** [comment]
Related: https://github.com/Project-OSRM/osrm-backend/issues/2055

**Comment by TheMarex** [comment]
I don't think it makes sense to create checkpoints in the middle of a tool. This complicates the control flow, and breaks with expected behavior (basically the cache invalidation problem).

If we need a checkpoint we should split the tool into two parts with defined control flows.

In this specific example we could split it apart easily. But I suspect rewriting the initial OSM extraction code to be in-memory would probably make it unnecessary to do so, since we could exploit parallelism for all operations, rather then be IO bound. See https://github.com/Project-OSRM/osrm-backend/issues/3767 for an actionable task for reducing the memory usage enough to make this feasible.
